Content.Client/Fax/UI/FaxBoundUi.cs

```diff
@@ -22,12 +28,57 @@ protected override void Open()
         _window.OpenCentered();
 
         _window.OnClose += Close;
+        _window.FileButtonPressed += OnFileButtonPressed;
         _window.CopyButtonPressed += OnCopyButtonPressed;
         _window.SendButtonPressed += OnSendButtonPressed;
         _window.RefreshButtonPressed += OnRefreshButtonPressed;
         _window.PeerSelected += OnPeerSelected;
     }
 
+    public async Task PrintFile()
+    {
+        //Open file select dialog
+        Stream? file;
+        var filters = new FileDialogFilters(new FileDialogFilters.Group("txt"));
+        try
+        {
+            file = await _fileDialogManager.OpenFile(filters);            
+        }
+        catch(IOException)
+        {
+            return;
+        }
+
+        //If UI gets closed of file is null return.
+        if(_window == null)
+            return;
+        if (_window.Disposed)
+            return;
+        if(file == null)
+            return;
+
+        //Read the file contents and raise event.
+        string content;
+        try
+        {
+            StreamReader reader = new StreamReader(file);
+            content = reader.ReadToEnd();
+            reader.Close();
+            file.Close();
+        }
+        catch(IOException)
+        {
+            return;
+        }
+
+        SendMessage(new FaxFileMessage(content.Substring(0, Math.Min(content.Length, 10000)), "printed paper", _window.OfficePaper));
+    }
+
+    private void OnFileButtonPressed()
+    {
+        PrintFile();
+    }
```

Comment:
This looks worse than before, you removed all of the using statements in lieu of closing (which won't handle disposals correctly) and some of the state handling became weirder.

You also still need to validate the input on the server for both the name and paper length.

Reply:
let them cook